### PR TITLE
Move to Zephyr 3.6.0

### PR DESCRIPTION
**Pull request by Deedone**

### CODE REVIEW

File: snippets/xen_dom0/boards/qemu_cortex_a53.overlay [hunk 1]
```diff
@@ -16,13 +16,15 @@
 	 * This node may differs on different setups, please check
 	 * following line in Xen boot log to set it right:
 	 * (XEN) Grant table range: 0x00000040200000-0x00000040240000
+	 * Also, add extended region 1:
+	 * (XEN) Extended region 1: 0x40000000->0x47e00000
```
Comment (firscity) [inline snippets/xen_dom0/boards/qemu_cortex_a53.overlay:20]: `boards: shields: add extended region description to dom0 shield` -> `snippets: add extended region description to dom0 snippets`

File: modules/nanopb/CMakeLists.txt [hunk 2]
```diff
@@ -6,15 +6,25 @@ if(CONFIG_NANOPB)
 
   set(NANOPB_DIR ${ZEPHYR_CURRENT_MODULE_DIR})
 
+  find_program(PROTOC protoc)
```
Comment (firscity) [inline modules/nanopb/CMakeLists.txt]: Are these commits spurious? If no, please provide argumentation in commit message
`b18abbe Revert "modules: nanopb: Add custom target for generated header files"`
`61fdde05 Revert "modules: nanopb: Add helper function to generate sources"`

Comment (Deedone) [reply modules/nanopb/CMakeLists.txt]: They break AOS protobuf generation, it's a workaround to fix build that I forgot about, will try to fix this properly

**Comment by xakep-amatop** [comment]
It would be nice to wait until I remake 3ee924e5203a08c77225f49734f7cd388b0a8504 and use the new version instead. From my side, the other changes are okay.

-----
P.S. pls, fix checkpatch issues

**Comment by Deedone** [comment]
Fixed spider board name for tests and samples

**Comment by Deedone** [comment]
Replaced `fs: use memset to initialize fs_file_t and fs_dir_t` with the equivalent upstream patch.

File: snippets/xen_dom0/boards/qemu_cortex_a53.overlay [hunk 3]
```diff
@@ -16,13 +16,15 @@
...
 	 *
 	 * Xen passes actual values for setup in domain device tree, but Zephyr
 	 * is not capable to parse and handle it in runtime.
 	 */
 	hypervisor: hypervisor@40200000 {
 		compatible = "xen,xen";
-		reg = <0x0 0x40200000 0x0 0x40000>;
+		reg = <0x0 0x40200000 0x0 0x40000 0x0 0x40400000 0x0 0x17C00000>;
```
Comment (GrygiriiS) [inline snippets/xen_dom0/boards/qemu_cortex_a53.overlay:27]: What was added here does not correspond to what is mentioned in comment above?

Comment (Deedone) [reply snippets/xen_dom0/boards/qemu_cortex_a53.overlay:27]: I am not the author of this change, @oleksiimoisieiev could you please take a look?

**Review by firscity — APPROVED** [review]

**Pull request merged**